internal/config/cgroupmanager/systemdmanager.go

```diff
@@ -0,0 +1,183 @@
+// returns the cgroup parent, cgroup path, and error.
+// it also checks there is enough memory in the given cgroup (4mb is needed for the runtime)
+func (*Systemdv1Manager) GetSandboxCgroupPath(sbParent, sbID string) (string, string, error) {
+	return getSandboxCgroupPathForSystemd(sbParent, sbID, cgroupMemorySubsystemMountPathV1, "memory.limit_in_bytes")
+}
+
+// MoveConmonToCgroup takes the container ID, cgroup parent, conmon's cgroup (from the config) and conmon's PID
+// It attempts to move conmon to the correct cgroup.
+func (*Systemdv1Manager) MoveConmonToCgroup(cid, cgroupParent, conmonCgroup string, pid int) string {
+	return moveConmonToSystemdCgroup(cid, cgroupParent, conmonCgroup, pid)
+}
+
+type Systemdv2Manager struct{}
+
+// Name returns the name of the cgroup manager (systemd)
+func (*Systemdv2Manager) Name() string {
+	return SystemdCgroupManager
+}
+
+// IsSystemd returns that it is a systemd cgroup manager
+func (*Systemdv2Manager) IsSystemd() bool {
+	return true
+}
+
+// GetContainerCgroupPath takes arguments sandbox parent cgroup and container ID and returns
+// the cgroup path for that containerID. If parentCgroup is empty, it
+// uses the default parent system.slice
+func (*Systemdv2Manager) GetContainerCgroupPath(sbParent, containerID string) string {
+	return getContainerCgroupPath(sbParent, containerID)
+}
+
+// systemd cgroups (v1 and v2) have no difference in creating the container cgroup path
+func getContainerCgroupPath(sbParent, containerID string) string {
+	parent := defaultSystemdParent
+	if sbParent != "" {
+		parent = sbParent
+	}
+	return parent + ":" + scopePrefix + ":" + containerID
+}
+
+// GetSandboxCgroupPath takes the sandbox parent, and sandbox ID. It
+// returns the cgroup parent, cgroup path, and error.
+// it also checks there is enough memory in the given cgroup (4mb is needed for the runtime)
+func (*Systemdv2Manager) GetSandboxCgroupPath(sbParent, sbID string) (string, string, error) {
+	return getSandboxCgroupPathForSystemd(sbParent, sbID, cgroupMemorySubsystemMountPathV2, "memory.max")
+}
+
+func getSandboxCgroupPathForSystemd(sbParent, sbID, memorySubsystemPath, memoryPath string) (string, string, error) {
+	if sbParent == "" {
+		return "", "", nil
+	}
+
+	if len(sbParent) <= 6 || !strings.HasSuffix(path.Base(sbParent), ".slice") {
+		return "", "", fmt.Errorf("cri-o configured with systemd cgroup manager, but did not receive slice as parent: %s", sbParent)
+	}
+
+	cgroupParent := convertCgroupFsNameToSystemd(sbParent)
+
+	if err := verifyCgroupHasEnoughMemory(cgroupParent, memorySubsystemPath, "memory.limit_in_bytes"); err != nil {
+		return "", "", err
+	}
+
+	cgPath := cgroupParent + ":" + scopePrefix + ":" + sbID
+
+	return cgroupParent, cgPath, nil
+}
+
+func verifyCgroupHasEnoughMemory(cgroupParent, memorySubsystemPath, memoryMaxFilename string) error {
+	// check memory limit is greater than the minimum memory limit of 4Mb
```

Comment:
Hmm, it was 12 MB above.

Also, this comment describes what the function does so it should go just before its declaration.

Either fix and move it, or drop it, either way is fine to me

Reply:
bye bye comment :wave: